File: android/src/main/java/com/reactnativechimesdk/data/VideoCollectionTile.java
package com.reactnativechimesdk.data;

import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.VideoTileState;

import java.util.Objects;

public class VideoCollectionTile {

  private final VideoTileState videoTileState;

  public VideoCollectionTile(VideoTileState videoTileState) {
    this.videoTileState = videoTileState;
  }

  public VideoTileState getVideoTileState() {
    return videoTileState;
  }

  @Override
  public boolean equals(Object o) {
    if (o == null) {
      return false;
    }
    if (this == o) {
      return true;
    }
    if (!(o instanceof VideoCollectionTile)) {
      return false;
    }
    VideoCollectionTile that = (VideoCollectionTile) o;
    return videoTileState.getAttendeeId().equals(that.videoTileState.getAttendeeId());
  }

  @Override
  public int hashCode() {
    return Objects.hash(videoTileState.getAttendeeId());
  }
}

File: android/src/main/java/com/reactnativechimesdk/data/VideoCollectionTile.kt
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */
package com.reactnativechimesdk.data

import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.DefaultVideoRenderView
import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.VideoTileState

data class VideoCollectionTile(
    val attendeeName: String,
    val videoTileState: VideoTileState
) {
    var videoRenderView: DefaultVideoRenderView? = null

    fun setRenderViewVisibility(visibility: Int) {
        videoRenderView?.visibility = visibility
    }
}
